### PR TITLE
`x/sync` / `x/merkledb` -- add `SyncableDB` interface

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

The syncer only needs a subset of the `MerkleDB` methods. This will make implementing the interface (i.e. via gRPC in tests) easier.

## How this works

Pretty straightforward -- break out a new `SyncableDB` and embed that in MerkleDB and the sync manager.

## How this was tested

Update existing UT

### CODE REVIEW

File: x/merkledb/trie.go [hunk 1]
```diff
@@ -13,7 +13,21 @@ import (
 
 var errNoNewRoot = errors.New("there was no updated root in change list")
 
+type MerkleRootGetter interface {
+	// GetMerkleRoot returns the merkle root of the Trie
+	GetMerkleRoot(ctx context.Context) (ids.ID, error)
+}
+
+type ProofGetter interface {
```
Comment (danlaine) [inline x/merkledb/trie.go:21]: Made these interfaces so we don't need to declare the same method in `ReadOnlyTrie` and `SyncableDB`

File: x/sync/syncmanager.go [hunk 2]
```diff
@@ -106,7 +106,7 @@ type StateSyncManager struct {
 }
 
 type StateSyncConfig struct {
-	SyncDB                merkledb.MerkleDB
+	SyncDB                merkledb.SyncableDB
```
Comment (danlaine) [inline x/sync/syncmanager.go:109]: Now `x/sync` depends on a smaller interface than `MerkleDB`. Yay!

Comment (darioush) [reply x/sync/syncmanager.go:109]: Should we move the new interface to the `sync` package?

Comment (danlaine) [reply x/sync/syncmanager.go]: moved to  `x/sync`

**Review by darioush — APPROVED** [review]

**Review by xinifinity — APPROVED** [review]

**Pull request merged**